platform/linux-generic/odp_timer.c

```diff
@@ -395,20 +436,25 @@ static odp_timer_pool_t timer_pool_new(const char *name,
 			odp_time_global_from_ns(nsec_per_scan);
 	}
 
+	/* Update the highest index for inline timer scan */
+	if (tp_idx > timer_global->highest_tp_idx)
+		timer_global->highest_tp_idx = tp_idx;
+
 	odp_ticketlock_unlock(&timer_global->lock);
 
-	if (!odp_global_rw->inline_timers) {
-		if (tp->param.clk_src == ODP_CLOCK_DEFAULT)
```

Comment:
The `odp_timer_pool_param_t.clk_src` is never checked in `odp_timer_pool_create()`. Using a wrong clock source doesn't seem to have any effect, but it would still be good to detect it.

Reply:
Fixed in v2